include/seqan3/alphabet/aminoacid/translation.hpp

```diff
@@ -0,0 +1,187 @@
+//     FLATWORM_MITOCHONDRIAL = 9,
+//     EUPLOTID,
+//     PROKARYOTE,
+//     ALT_YEAST,
+//     ASCIDIAN_MITOCHONDRIAL,
+//     ALT_FLATWORM_MITOCHONDRIAL,
+//     BLEPHARISMA,
+//     CHLOROPHYCEAN_MITOCHONDRIAL,
+//     TREMATODE_MITOCHONDRIAL = 21,
+//     SCENEDESMUS_MITOCHONDRIAL,
+//     THRAUSTOCHYTRIUM_MITOCHONDRIAL,
+//     PTEROBRANCHIA_MITOCHONDRIAL,
+//     GRACILIBACTERIA
+};
+}
+
+#include <seqan3/alphabet/aminoacid/translation_details.hpp>
+
+namespace seqan3
+{
+/*!\brief Translate one nucleotide triplet into single amino acid (single nucleotide interface)
+ *
+ * \param n1 First nucleotide in triplet
+ * \param n2 Second nucleotide in triplet
+ * \param n3 Third nucleotide in triplet
+ *
+ * \details
+ *
+ * Translates single nucleotides into amino acid according to given genetic code.
+ *
+ * \par Complexity
+ *
+ * Constant.
+ *
+ * \par Exceptions
+ *
+ * No-throw guarantee.
+*/
+template <nucleotide_concept nucl_type, genetic_code gc = genetic_code::CANONICAL>
+constexpr aa27 translate_triplet(nucl_type const & n1, nucl_type const & n2, nucl_type const & n3) noexcept
+{
+    return seqan3::detail::translation_table<nucl_type, gc>::VALUE[n1.to_rank()][n2.to_rank()][n3.to_rank()];
+}
+
+/*!\brief Translate one nucleotide triplet into single amino acid (tuple interface)
+ * \tparam tuple_type The type of input tuple
+ * \param input_tuple Triplet of nucleotides that should be converted to amino acid
+ * \details
+ *
+ * Translates std::tuple or std::array with 3 nucleotides into amino acid according to given genetic code.
+ *
+ * \par Complexity
+ *
+ * Constant.
+ *
+ * \par Exceptions
+ *
+ * No-throw guarantee.
+*/
+template <typename tuple_type, genetic_code gc = genetic_code::CANONICAL>
+    requires std::tuple_size<tuple_type>::value == 3 &&
+             nucleotide_concept<std::tuple_element_t<0, tuple_type>> &&
+             nucleotide_concept<std::tuple_element_t<1, tuple_type>> &&
+             nucleotide_concept<std::tuple_element_t<2, tuple_type>>
+constexpr aa27 translate_triplet(tuple_type const & input_tuple) noexcept
+{
+    return translate_triplet(std::get<0>(input_tuple), std::get<1>(input_tuple), std::get<2>(input_tuple));
+}
+
+/*!\brief Translate one nucleotide triplet into single amino acid (range interface)
+ * \tparam range_type The type of input range
+ * \param input_range Range of three nucleotides that should be converted to amino acid
+ *
+ * \details
+ *
+ * Translates range with 3 nucleotides into amino acid according to given genetic code.
+ *
+ * \par Complexity
+ *
+ * Constant.
+ *
+ * \par Exceptions
+ *
+ * Strong exception guarantee (never modifies data).
+*/
+template <typename range_type, genetic_code gc = genetic_code::CANONICAL>
+    requires input_range_concept<range_type> &&
+             nucleotide_concept<std::decay_t<ranges::range_reference_t<std::decay_t<range_type>>>>
+constexpr aa27 translate_triplet(range_type && input_range)
+{
+    assert(ranges::size(input_range) == 3);
```

Comment:
ranges::size is only defined on sized_range_concept which we do not require here. I would instead check after each of the assignments below:
```cpp
assert(n1 != ranges::end(input_range));
```